src/JointTrajectoryControllerBase.cpp

```diff
@@ -297,8 +312,24 @@ void JointTrajectoryControllerBase::updateStep(const ros::Time& time,
 
   for (size_t idof = 0; idof < mAdapters.size(); ++idof)
   {
-    mAdapters[idof]->update(time, period, mActualPosition[idof],
-                            mDesiredPosition[idof], mActualVelocity[idof],
+    // Check for SO2
+    auto actualPos = mActualPosition[idof];
+    auto desiredPos = mDesiredPosition[idof];
+    auto jointSpace = mControlledSpace->getJointSpace(idof);
+    auto r1Joint = std::dynamic_pointer_cast<const R1Joint>(jointSpace);
+    if(!r1Joint) {
+      desiredPos = std::fmod(desiredPos, 2.0 * M_PI);
+      actualPos = std::fmod(actualPos, 2.0 * M_PI);
+      if(desiredPos - actualPos > M_PI) {
+        actualPos += 2*M_PI;
```

Comment:
Is it always safe to update `actualPos`, not `desiredPos`? 

`JointVelocityAdapter` and `JointEffortAdapter` look at the difference between the two but `JointPositionAdapter` looks only at the `desiredPos`.
https://github.com/personalrobotics/rewd_controllers/blob/master/src/JointAdapter.cpp#L48

I think we need to update `desiredPos` to be within M_PI around `actualPos`.